runelite-client/src/test/java/net/runelite/client/util/ColorUtilTest.java

```diff
@@ -123,8 +126,66 @@ public void colorLerp()
 	public void colorToHexCode()
 	{
 		COLOR_HEXSTRING_MAP.forEach((color, hex) ->
-		{
-			assertEquals(hex, ColorUtil.colorToHexCode(color));
-		});
+			assertEquals(hex, ColorUtil.colorToHexCode(color)));
+	}
+
+	@Test
+	public void colorToAlphaHexCode()
+	{
+		COLOR_ALPHA_HEXSTRING_MAP.forEach((color, hex) ->
+			assertEquals(hex, ColorUtil.colorToAlphaHexCode(color)));
+	}
+
+	@Test
+	public void isHex()
+	{
+		COLOR_HEXSTRING_MAP.forEach((_color, hex) ->
+			assertTrue(ColorUtil.isHex(hex)));
+	}
+
+	@Test
+	public void isAlphaHex()
+	{
+		COLOR_ALPHA_HEXSTRING_MAP.forEach((_color, hex) ->
+			assertTrue(ColorUtil.isAlphaHex(hex)));
+	}
+
+	@Test
+	public void fromString()
+	{
+		String WHITE_MAX_ALPHA = "-1";
+		String WHITE_ZERO_ALPHA = "0xffffff";
+		String TOO_LARGE = "0xffffffff";
+		String INVALID_FORMAT = "ffffff";
+
+		assertEquals(Color.WHITE, ColorUtil.fromString(WHITE_MAX_ALPHA));
+		assertEquals(new Color(0xffffff, true), ColorUtil.fromString(WHITE_ZERO_ALPHA));
+		assertNotEquals(Color.WHITE, ColorUtil.fromString(WHITE_ZERO_ALPHA));
+
+		assertNull(ColorUtil.fromString(TOO_LARGE));
+		assertNull(ColorUtil.fromString(INVALID_FORMAT));
+	}
+
+	@Test
+	public void fromHex()
+	{
+		assertEquals(Color.BLACK, ColorUtil.fromHex("0x000000"));
+		assertEquals(Color.BLACK, ColorUtil.fromHex("#000000"));
+		assertEquals(Color.BLACK, ColorUtil.fromHex("000000"));
+
+		assertEquals(Color.BLACK, ColorUtil.fromHex("0x0"));
+		assertEquals(Color.BLACK, ColorUtil.fromHex("#0"));
+
+		assertEquals(Color.WHITE, ColorUtil.fromHex("0xFFFFFFFF"));
+
+		assertNull(ColorUtil.fromHex("0x"));
+		assertNull(ColorUtil.fromHex("########"));
+		assertNull(ColorUtil.fromHex("0xABCDEFGH"));
+	}
+
+	@Test
+	public void fromObject()
+	{
+		assertEquals(Color.RED, ColorUtil.fromObject(""));
```

Comment:
This test isn't useful. We don't want to guarantee that an empty string will yield `Color.RED`. It would be much more helpful to assert the result for a variety of objects is not null.

Reply:
I've replaced this test with what you've suggested.